gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java

```diff
@@ -622,13 +622,29 @@ public ImageEntity getPicture(String apiId) {
     @Override
     public void deleteViewFromAPIs(final String viewId) {
         findAll().forEach(api -> {
-            if (api.getViews()
+            if (api.getViews() != null && api.getViews()
                     .removeIf(view -> view.equals(viewId))) {
```

Comment:
You are doing a remove of view twice : once by the removeIf, and on the remove() method...